src/event/log_event.rs

```diff
@@ -20,6 +20,17 @@ pub struct LogEvent {
 }
 
 impl LogEvent {
+    pub fn new_with_metadata(metadata: EventMetadata) -> Self {
+        Self {
+            fields: Default::default(),
+            metadata,
+        }
+    }
+
+    pub fn deconstruct(self) -> (BTreeMap<String, Value>, EventMetadata) {
```

Comment:
I don't have a strong opinion either way, but I have seen `into_parts` as a relatively common name for this kind of method.

Reply:
That makes sense, I'll do that.